### PR TITLE
add message list method and tests

### PR DESCRIPTION
**Pull request by natebrennand**
adds list and filter support for Messages

also an example program

### CODE REVIEW

File: sms/sms.go [hunk 1]
```diff
@@ -97,3 +98,38 @@ func (act SmsAccount) Get(sid string) (Message, error) {
...
+		v.Set("DateSent", f.DateSent.Format(common.GMTTimeLayout))
+	}
+	return strings.NewReader(v.Encode())
+}
+
+func (f Filter) Validate() error {
+	return nil
+}
+
+func (act SmsAccount) getList(destUrl string, f Filter, resp *MessageList) error {
```
Comment (chelpu) [inline sms/sms.go:131]: This might be a silly question, but is f ever being used?

Comment (natebrennand) [inline sms/sms.go:131]: whoops, updating now

**Comment by natebrennand** [comment]
@chelpu, updated.
tested with my example and it works

**Comment by natebrennand** [comment]
weird, tests pass locally fine but travis failed, restarting the build

**Pull request merged**